### PR TITLE
[CIKM22Competition] Allow user to specify path for prediction saving

### PR DESCRIPTION
**Pull request by DavdGao**
as the title says

### CODE REVIEW

File: federatedscope/gfl/trainer/graphtrainer.py [hunk 1]
```diff
@@ -84,14 +84,14 @@ def _hook_on_batch_forward_flop_count(self, ctx):
...
+    def save_prediction(self, path, client_id, task_type):
         y_inds, y_probs = self.ctx.test_y_inds, self.ctx.test_y_prob
-        os.makedirs('prediction', exist_ok=True)
+        os.makedirs(path, exist_ok=True)
 
         # TODO: more feasible, for now we hard code it for cikmcup
         y_preds = np.argmax(y_probs, axis=-1) if 'classification' in task_type.lower() else y_probs
 
-        with open('prediction/prediction.csv', 'a') as file:
+        with open(path+'/prediction.csv', 'a') as file:
```
Comment (joneswong) [inline federatedscope/gfl/trainer/graphtrainer.py:94]: please use `os.path.join` to concat strings into a valid path.

**Review by joneswong — CHANGES_REQUESTED** [review]
Please ensure the consistency between this change and your guidance.

**Comment by DavdGao** [comment]
> 

Done

File: federatedscope/gfl/trainer/graphtrainer.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/gfl/trainer/graphtrainer.py]: Done

**Comment by DavdGao** [comment]
> Please ensure the consistency between this change and your guidance.

It has been checked

**Review by joneswong — APPROVED** [review]
approved.

**Comment by joneswong** [comment]
this pr is related to #249

**Pull request merged**